src/ape_ethereum/ecosystem.py

```diff
@@ -660,7 +662,16 @@ def create_transaction(self, **kwargs) -> TransactionAPI:
         kwargs["gas"] = kwargs.pop("gas_limit", kwargs.get("gas"))
 
         if "value" in kwargs and not isinstance(kwargs["value"], int):
-            kwargs["value"] = self.conversion_manager.convert(kwargs["value"], int)
+            value = kwargs["value"] or 0  # Convert None to 0.
+            kwargs["value"] = self.conversion_manager.convert(value, int)
+
+        # This causes problems in pydantic for some reason.
+        if "gas_price" in kwargs and kwargs["gas_price"] is None:
+            del kwargs["gas_price"]
+
+        # None is not allowed, the user likely means `b""`.
+        if "data" in kwargs and kwargs["data"] is None:
+            kwargs["data"] = b""
 
         return txn_class(**kwargs)
```

Comment:
note: can't remove all None values because for some reason, the validators get all fudged up with gas_limit when i do... so every field is handled case by case